In [10]:
import pandas as pd
import numpy as np

In [11]:
data = pd.DataFrame()

data['play'] =['No','No','Yes','Yes','Yes','No','Yes','No','Yes','Yes','Yes','Yes','Yes','No']

In [12]:
#attributes
data['weather'] = ['Sunny','Sunny','Overcast','Rainy','Rainy','Rainy','Overcast','Sunny','Sunny',
                   'Rainy','Sunny','Overcast','Overcast','Rainy']
data['temp'] = ['Hot','Hot','Hot','Mild','Cool','Cool','Cool','Mild','Cool','Mild','Mild','Mild','Hot','Mild']
data['windy'] = ['Strong','Strong','Weak','Weak','Strong','Weak','Strong','Weak',
                 'Weak','Strong','Weak','Weak','Strong','Weak']

In [16]:
data

play   weather  temp   windy
0    No     Sunny   Hot  Strong
1    No     Sunny   Hot  Strong
2   Yes  Overcast   Hot    Weak
3   Yes     Rainy  Mild    Weak
4   Yes     Rainy  Cool  Strong
5    No     Rainy  Cool    Weak
6   Yes  Overcast  Cool  Strong
7    No     Sunny  Mild    Weak
8   Yes     Sunny  Cool    Weak
9   Yes     Rainy  Mild  Strong
10  Yes     Sunny  Mild    Weak
11  Yes  Overcast  Mild    Weak
12  Yes  Overcast   Hot  Strong
13   No     Rainy  Mild    Weak

In [17]:
new = pd.DataFrame()

# สร้างข้อมูลที่เป็น new data 
new['weather'] = ['rainy']
new['temp'] = ['cool']
new['windy'] = ['weak']

# View the data 
new

weather  temp windy
0   rainy  cool  weak

# สภาพอากาศแบบนี้จะมีคนเล่นเทนนิสหรือไม่

In [22]:
# นับจำนวน yes
n_yes = len(data[data['play'] == 'Yes'])  #len(object) เป็น function ใช้นับจำนวน object

# นับจำนวน no
n_no = len(data[data['play'] == 'No']) 

# นับจำนวนที่เล่นทั้งหมด
total_play = len(data['play'])

print(n_yes,n_no,total_play)

9 5 14


In [21]:
n_rain_yes = len(data[(data['weather'] == 'Rainy') & (data['play'] == 'Yes')])
n_rain_yes

3

In [23]:
n_cool_yes = len(data[(data['temp'] == 'Cool') & (data['play'] == 'Yes')])
n_cool_yes

3

In [24]:
n_weak_yes = len(data[(data['windy'] == 'Weak') & (data['play'] == 'Yes')])
n_weak_yes

5

In [31]:
n_rain_no = len(data[(data['weather'] == 'Rainy') & (data['play'] == 'No')])
n_rain_no

2

In [29]:
n_cool_no = len(data[(data['temp'] == 'Cool') & (data['play'] == 'No')])
n_cool_no

1

In [30]:
n_weak_no = len(data[(data['windy'] == 'Weak') & (data['play'] == 'No')])
n_weak_no

3

In [33]:
p_yes = n_yes/total_play 
p_no = n_no/total_play

p_rain_yes = n_rain_yes/n_yes
p_rain_yes

0.3333333333333333

In [34]:
p_cool_yes = n_cool_yes/n_yes
p_cool_yes

0.3333333333333333

In [35]:
p_weak_yes = n_weak_yes/n_yes
p_weak_yes

0.5555555555555556

In [36]:
p_rain_no = n_rain_no/n_no
p_rain_no

0.4

In [37]:
p_cool_no = n_cool_no/n_no
p_cool_no

0.2

In [38]:
p_weak_no = n_weak_no/n_no
p_weak_no

0.6

In [47]:
p_rain_cool_weak_yes = p_rain_yes * p_cool_yes * p_weak_yes * p_yes
p_rain_cool_weak_no = p_rain_no * p_cool_no * p_weak_no * p_no



In [48]:
print(p_rain_yes,p_cool_yes,p_weak_yes,p_yes)

0.3333333333333333 0.3333333333333333 0.5555555555555556 0.6428571428571429


In [49]:
print(p_rain_no,p_cool_no,p_weak_no,p_no)

0.4 0.2 0.6 0.35714285714285715


In [50]:
print('p_rain_cool_weak_yes = ', p_rain_cool_weak_yes)
print('p_rain_cool_weak_no = ', p_rain_cool_weak_no)

p_rain_cool_weak_yes =  0.03968253968253968
p_rain_cool_weak_no =  0.017142857142857147


##  ค่า P(play = no|new) (=0.01714) มีค่ามากกว่า P(play = yes|new) (=0.03968) ดังนั้นโมเดลจึงทำนายว่าข้อมูลตัวอย่างนี้มีค่าคลาส play = yes เล่น tennis